modular_skyrat/master_files/code/modules/pain_shock/status_indicator.dm

```diff
@@ -0,0 +1,198 @@
+#define VIS_STATUS			24
+
+#define VIS_COUNT			24 //Must be highest number from above.
+#define STATUS_LAYER -2.1
+#define STATUS_INDICATOR_Y_OFFSET 2 // Offset from the edge of the icon sprite, so 32 pixels plus whatever number is here.
+#define STATUS_INDICATOR_ICON_X_SIZE 0 // Don't need to care about the Y size due to the origin being on the bottom side.
+#define STATUS_INDICATOR_ICON_MARGIN 0 // The space between two status indicators. We don't do this with the current icons.
+
+/datum/preference/toggle/enable_status_indicators
+	category = PREFERENCE_CATEGORY_GAME_PREFERENCES
+	savefile_key = "enable_status_indicators"
+	savefile_identifier = PREFERENCE_PLAYER
+
+/datum/preference/toggle/enable_status_indicators/create_default_value()
+	return TRUE
+
+/datum/preference/toggle/enable_status_indicators/apply_to_client(client/client, value)
+	. = ..()
+	var/atom/movable/screen/plane_master/runechat/status/status = locate() in client?.screen
+	if(!status)
+		return
+	if(value)
+		status.alpha = 255
+	else
+		status.alpha = 0
+/mob
+	var/status_enabled = TRUE
+/mob/living
+	var/list/status_indicators = null // Will become a list as needed.
+
+/* /mob/living/carbon/proc/is_critical()
+	if(HAS_TRAIT(src, TRAIT_CRITICAL_CONDITION))
+		return TRUE */
+
+/mob/living/proc/is_weakened()
+	if(HAS_TRAIT_FROM(src, TRAIT_FLOORED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) || has_status_effect(/datum/status_effect/incapacitating/knockdown))
+		return TRUE
+
+/mob/living/proc/is_stunned()
+	if(HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) || HAS_TRAIT_FROM(src, TRAIT_IMMOBILIZED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) || HAS_TRAIT_FROM(src, TRAIT_IMMOBILIZED, CHOKEHOLD_TRAIT))
+		return TRUE
+
+/mob/living/proc/is_paralyzed()
+	if(HAS_TRAIT_FROM(src, TRAIT_FLOORED, CHOKEHOLD_TRAIT) || HAS_TRAIT(src, TRAIT_CRITICAL_CONDITION) || HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, STAMINA) || HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) && HAS_TRAIT_FROM(src, TRAIT_IMMOBILIZED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) && HAS_TRAIT_FROM(src, TRAIT_FLOORED, TRAIT_STATUS_EFFECT(STAT_TRAIT)))
+		return TRUE
+
+/mob/living/proc/is_unconcious()
+	if(HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
+		return TRUE
+
+/mob/living/proc/is_confused()
+	if(has_status_effect(/datum/status_effect/confusion))
+		return TRUE
+
+/* /mob/living/carbon/proc/is_blind_status()
+	if(HAS_TRAIT_FROM(src, TRAIT_BLIND, STATUS_EFFECT_TRAIT) || HAS_TRAIT_FROM(src, TRAIT_BLIND, EYES_COVERED) && !HAS_TRAIT_FROM(src, TRAIT_BLIND, QUIRK_TRAIT)) // We don't want permanently blind users (those who took the trait) to have this forever.
+		return TRUE */ // Disabled, uncomment if we want this.
+
+/mob/living/carbon/death(gibbed) // On death, we clear the indiciators
+	..() // Call the TG death. Do not . = ..()!
+	for(var/iteration in status_indicators) // When we die, clear the indicators.
+		remove_status_indicator(icon_state) // The indicators are named after their icon_state and type
```

Comment:
Why are you removing them only on `/mob/living/carbon` when you're adding them to `/mob/living`?